src/engine/JoinHelpers.h

```diff
@@ -0,0 +1,122 @@
+//   Copyright 2025, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#ifndef JOINHELPERS_H
+#define JOINHELPERS_H
+
+#include <array>
+#include <optional>
+#include <vector>
+
+#include "engine/idTable/IdTable.h"
+#include "util/Generators.h"
+#include "util/JoinAlgorithms/JoinColumnMapping.h"
+#include "util/TypeTraits.h"
+
```

Comment:
Wrap those into a `namespace qlever::joinHelpers` or something whiich you still can `using namespace` in the `.cpp` files, 
and use the opportunity to complete the documentation of the missing elements and nicely format this header (empty lines between a definition and the docstring of the next thing.